reference/forms/types/choice.rst

```diff
@@ -83,40 +87,98 @@ code, while the **key** is what will be shown to the user.
     and will be removed in 3.0. To read about the old API, read an older version of
     the docs.
 
+.. note::
+
+    Pre selected choices will depend on the **data** passed to the field and
+    the values of the ``choices`` option. However submitted choices will depend
+    on the **string** matching the **choice**. In the example above, the default
+    values are incrementing integers because ``null`` cannot be casted to string.
```

Comment:
In fact, I think it's a bug ;) we can show `"", "1", "0"` attr values for `array(null, true, false)` choices and be more consistent.

Trying to fix that in https://github.com/symfony/symfony/pull/21160